### PR TITLE
Bug 1722214: thread safety fix for samples operator condidtion update…

### PR DESCRIPTION
**Pull request by gabemontero**
…; prevent duplicate entries in condition array

/assign @adambkaplan 
@openshift/openshift-team-developer-experience fyi

### CODE REVIEW

File: pkg/apis/samples/v1/types.go [hunk 1]
```diff
@@ -37,6 +38,12 @@ type Config struct {
 	Status ConfigStatus `json:"status,omitempty"`
 }
 
+var (
+	// conditionUpsertLock helps us avoid duplicat entries in the condition array when mutliple
```
Comment (adambkaplan) [inline pkg/apis/samples/v1/types.go]: nit - duplicat/duplicate

Comment (gabemontero) [reply pkg/apis/samples/v1/types.go]: thanks @adambkaplan typo fixed

**Pull request merged**